executor/distsql.go

```diff
@@ -54,6 +54,21 @@ func resultRowToRow(t table.Table, h int64, data []types.Datum, tableAsName *mod
 	return &Row{Data: data, RowKeys: []*RowKeyEntry{entry}}
 }
 
+// calculateGeneratedColumns calculates all virtual generated columns, with such arguments:
+//  cols is the collection of columns of vs;
+//  gvs's keys are index in cols, and the values are generation expressions.
+func calculateGeneratedColumns(cols []*model.ColumnInfo, gvs map[int]expression.Expression, vs []types.Datum) (err error) {
+	for i, col := range cols {
+		if len(col.GeneratedExprString) != 0 && !col.GeneratedStored {
+			vs[i], err = gvs[col.Offset].Eval(vs)
```

Comment:
no, all generation expressions will be stored in `enValues` of `XSelectTableExec`.